kobo/django/xmlrpc/decorators.py

```diff
@@ -40,40 +48,76 @@ def _new_func(request, *args, **kwargs):
     return _new_func
 
 
-def login_required(func):
-    def _new_func(request, *args, **kwargs):
-        if not call_if_callable(request.user.is_authenticated):
-            raise PermissionDenied("Login required.")
-        return func(request, *args, **kwargs)
-
-    _new_func.__name__ = func.__name__
-    _new_func.__doc__ = func.__doc__
-    _new_func.__dict__.update(func.__dict__)
-    return _new_func
-
-
-def admin_required(func):
-    def _new_func(request, *args, **kwargs):
-        if not request.user.is_superuser:
-            raise PermissionDenied("Admin only.")
-        return func(request, *args, **kwargs)
-
-    _new_func.__name__ = func.__name__
-    _new_func.__doc__ = func.__doc__
-    _new_func.__dict__.update(func.__dict__)
-    return _new_func
-
-
-def validate_user(func):
-    def _new_func(request, user_id, *args, **kwargs):
-        if request.user.id != user_id:
-            raise SuspiciousOperation("UserID doesn't match logged in user.")
-        return func(request, user_id, *args, **kwargs)
-
-    _new_func.__name__ = func.__name__
-    _new_func.__doc__ = func.__doc__
-    _new_func.__dict__.update(func.__dict__)
-    return _new_func
+# decorator for required login with optional logger parameter
+def login_required(_func=None, *, logger=None):
+    def inner_login_required(func):
+        def _new_func(request, *args, **kwargs):
+            if not call_if_callable(request.user.is_authenticated):
+                if logger:
+                    user_ip = get_client_ip(request)
+                    logger.info("User with IP %s attempted admin operation.", user_ip)
```

Comment:
It's not an admin operation, it's just a method which requires authentication and they haven't authenticated.

Maybe a message like this would be better:

```
"Unauthenticated user with IP %s attempted to use authenticated method %s", user_ip, func.__name__
```

Reply:
Thanks. Changed.